### PR TITLE
feat(integration-git): init git integration storage layer and git client

### PR DESCRIPTION
**Pull request by LuckyPickleZZ**
#### What type of PR is this?
type-feature

#### What this PR does / why we need it:
init git integration storage and git client layer.

### CODE REVIEW

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
+CREATE TABLE IF NOT EXISTS `integration_git_repository` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `update_time` datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  `creator_id` bigint(20) NOT NULL COMMENT 'creator user id, references iam_user.id',
+  `organization_id` bigint(20) NOT NULL COMMENT 'organization id, references iam_organization.id',
+  `project_id` bigint(20) NOT NULL COMMENT 'project id, references collaboration_project.id',
```
Comment (yhilmare) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:7]: a git repo should not belong to any project once created~
in another word, `project_id` can be null

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  `description` varchar(512) DEFAULT NULL COMMENT 'description',
+  `name` varchar(512) NOT NULL COMMENT 'repository name',
+  `provider_type` varchar(32) NOT NULL COMMENT 'git provider type',
+  `provider_url` varchar(128) NOT NULL COMMENT 'git provider url',
+  `ssh_url` varchar(512) NOT NULL COMMENT 'ssh clone url, like git@github.com:xxx/xxx.git',
+  `clone_url` varchar(512) NOT NULL COMMENT 'https clone url, like https://github.com/xxx/xxx.git',
+  `email` varchar(128) NOT NULL COMMENT 'user email for commit',
+  `token` varchar(512) NOT NULL COMMENT 'person access token',
```
Comment (yhilmare) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:15]: token is a private ingo, just like password, we should encrypt it, just like what we do for connection config~

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `create_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  `update_time` datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  `description` varchar(512) DEFAULT NULL COMMENT 'description',
+  `repo_id` bigint(20) NOT NULL COMMENT 'git repository id, references integration_git_repository.id',
+  `state` varchar(32) NOT NULL COMMENT 'local repository state',
+  `branch` varchar(512) NOT NULL COMMENT 'last edit branch',
+  `last_commit_id` varchar(64) DEFAULT NULL COMMENT 'git commit revision number',
+  `diff_patch_storage` varchar(512) DEFAULT NULL COMMENT 'storage information of patch file',
+  `user_id` bigint(20) NOT NULL COMMENT 'user id, references iam_user.id',
```
Comment (yhilmare) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:30]: we'd better use a more specific field name.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/git/model/GitRepositoryMapper.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import com.oceanbase.odc.service.encryption.EncryptionFacade;
+
+import lombok.NonNull;
+
+/**
+ * @author: liuyizhuo.lyz
+ * @date: 2024/7/29
+ */
+@Component
+public class GitRepositoryMapper {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/git/model/GitRepositoryMapper.java:33]: `org.mapstruct.Mapper`
`org.mapstruct.Mapping`

will help do this~

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql (hunk 3 above)
Comment (LuckyPickleZZ) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:30]: Sorry I think this name is appropriate. Every user would own a copy of the repo, and this table storage the stages of these copies. So this column does indeed represent the user id

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql (hunk 2 above)
Comment (LuckyPickleZZ) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:15]: Okay, I will encrypt it in service layer in the future.

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql (hunk 1 above)
Comment (LuckyPickleZZ) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql]: done~

File: server/odc-service/src/main/java/com/oceanbase/odc/service/git/model/GitRepositoryMapper.java (hunk 4 above)
Comment (LuckyPickleZZ) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/git/model/GitRepositoryMapper.java]: Thanks, I've replaced this with it.

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql (hunk 3 above)
Comment (yhilmare) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:30]: ok

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql (hunk 2 above)
Comment (yhilmare) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:15]: okay, you can take a look at the table `connect_connection`, there is several columns related to encryption, just like `cipher`, `salt`

Comment (LuckyPickleZZ) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:15]: Thanks for your remind! I added `salt` into the entity but missed to add into ddl.

**Review by yhilmare — APPROVED** [review]
lgtm

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  `update_time` datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  `creator_id` bigint(20) NOT NULL COMMENT 'creator user id, references iam_user.id',
+  `organization_id` bigint(20) NOT NULL COMMENT 'organization id, references iam_organization.id',
+  `project_id` bigint(20) DEFAULT NULL COMMENT 'project id, references collaboration_project.id',
+  `description` varchar(512) DEFAULT NULL COMMENT 'description',
+  `name` varchar(512) NOT NULL COMMENT 'repository name',
+  `provider_type` varchar(32) NOT NULL COMMENT 'git provider type',
+  `provider_url` varchar(128) NOT NULL COMMENT 'git provider url',
+  `ssh_url` varchar(512) NOT NULL COMMENT 'ssh clone url, like git@github.com:xxx/xxx.git',
+  `clone_url` varchar(512) NOT NULL COMMENT 'https clone url, like https://github.com/xxx/xxx.git',
```
Comment (yizhouxw) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:13]: ssh_address , due ssh syntax is not an url 
http_address, due we have ssh as clone address also

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  `email` varchar(128) NOT NULL COMMENT 'user email for commit',
+  `token` varchar(512) NOT NULL COMMENT 'person access token',
```
Comment (yizhouxw) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:15]: person_access_token,

or 

personal_access_token ?

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  `salt` varchar(32) DEFAULT NULL COMMENT 'used to connect the random value used by the encryption and decryption algorithm of the secret field',
+  PRIMARY KEY (`id`),
+  KEY `integration_git_repository_project_id` (`project_id`)
```
Comment (yizhouxw) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:18]: index by organization_id+project_id

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+) COMMENT = 'integration git repository';
+
+CREATE TABLE IF NOT EXISTS `integration_git_repository_stage` (
```
Comment (yizhouxw) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:21]: organization_id column always required for multi tenant architecture

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  `update_time` datetime DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  `description` varchar(512) DEFAULT NULL COMMENT 'description',
+  `repo_id` bigint(20) NOT NULL COMMENT 'git repository id, references integration_git_repository.id',
+  `state` varchar(32) NOT NULL COMMENT 'local repository state',
+  `branch` varchar(512) NOT NULL COMMENT 'last edit branch',
+  `last_commit_id` varchar(64) DEFAULT NULL COMMENT 'git commit revision number',
+  `diff_patch_storage` varchar(512) DEFAULT NULL COMMENT 'storage information of patch file',
+  `user_id` bigint(20) NOT NULL COMMENT 'user id, references iam_user.id',
+  PRIMARY KEY (`id`),
+  KEY `integration_git_edit_history_repo_id_user_id` (`repo_id`, `user_id`)
```
Comment (yizhouxw) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:33]: so as the index, contains organization_id

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql (hunk 5 above)
Comment (LuckyPickleZZ) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql]: okay, renamed these fields.

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql (hunk 6 above)
Comment (LuckyPickleZZ) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql]: renamed to personal_access_token

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql (hunk 7 above)
Comment (LuckyPickleZZ) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql]: done~

File: server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql (hunk 8 above)
Comment (LuckyPickleZZ) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_3_2_1__add_integration_git.sql:21]: done

**Review by yizhouxw — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**